chall08/vkurkela.c

```diff
@@ -0,0 +1,40 @@
+int ft_ie_except_after_c(char *str)
+{
+    int i;
+    int len;
+    int exc;
+    
+    i = 0;
+    len = 0;
+    exc = 1;
+    while (str[i++] != '\0')
+        len++;
+    i = 0;
+    while (str[i] != '\0')
+    {
+        if (str[i] == 'e' && (len - (i + 1)) >= 1)
+        {
+            if (i != 0)
+            {
+                if (str[i - 1] == 'c' && str[i + 1] == 'i')
+                    return (1);
```

Comment:
Returning at this point means that the rest of the string will not be validated. Instead you could return 0 when the condition doesn't fill.

Reply:
Thanks Liisa for the feedback! :) Yeah so my idea for here was to return 1 immediately because when this condition fills I am sure it follows the rule and I don't need to validate rest of the string. Same applies to other cases where I'm return 1 straight away. So I was wondering is there some reason why would you validate rest of the string?

Reply:
If the string has more than one combination of ei or ie you need to check if the rest of them follow the rule as well to confirm that the whole string follows the rule. Returning 0 at the first instance would be ok, since you know that the string does not follow the rule.

Reply:
that is true! I wasn't thinking clear enough. I should have return 0 if condition doesn't fill